src/integrator_utils.jl

```diff
@@ -275,3 +275,82 @@ function advance_ode_integrator!(integrator::DDEIntegrator)
     # update prev_idx to index of t and u(t) in solution
     integrator.prev_idx = length(integrator.sol.t)
 end
+
```

Comment:
Could you explain the strategy that's going on in this file? I am not getting it.

Reply:
The idea is: if we hit a discontinuity (this is checked in https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/master/src/integrators/integrator_utils.jl#L349-L350), then we remove the discontinuity and maybe additional discontinuities at the current time point (and maybe also discontinuities and time stops that are close by...) in `handle_discontinuities!`. The order of the discontinuity at the current time point is defined as the lowest order of all these discontinuities. If the problem is not neutral, we will only add additional discontinuities (in `add_next_discontinuities!`) if this order is below the order of the algorithm (which in future should also depend on the used interpolants, I guess). If we add discontinuities, we just add discontinuities of the next order caused by constant lags (these we can calculate explicitly and just add them to `d_discontinuities` and `tstops`) and add the current discontinuity to `tracked_discontinuities` which is the array of old discontinuities that are checked by the `DiscontinuityCallback`.

Reply:
Got it, I'll read with this in mind. Could this be commented?

Reply:
I tried to add comments to all steps of this algorithm, should this summary be added as well?

Reply:
Yeah, I wasn't able to catch on a first read. I didn't get how things were aggregated together and why the minimum order, but when you lay it out like this it makes more sense.